fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -137,24 +136,24 @@ def fit(self, X, y, aux_data=None, **kwargs):
         # Prep the quality metric
         self.quality_metric.set_data(X, y_vector, A)
 
-        if isinstance(self.disparity_metric, ConditionalOpportunity):
+        if isinstance(self.disparity_metric, ClassificationMoment):
             # We have a classification problem
             # Need to make sure that y is binary (for now)
             unique_labels = np.unique(y_vector)
             if not set(unique_labels).issubset({0, 1}):
                 raise RuntimeError(self._MESSAGE_Y_NOT_BINARY)
 
         # Prep the disparity metric and objective
-        self.disparity_metric.init(X, A, y_vector)
+        self.disparity_metric.load_data(X, y_vector, **kwargs)    # group_membership=A)
```

Comment:
Don't need the end of this line?

Reply:
This is intentional. "fit" shouldn't be doing anything special with kwargs and just pass them on to the constraints (disparity_metric) and the objective. for the time being, we are doing some checks of sensitive_features inside "fit", but those kinds of checks should happen inside disparity_metric.